pkg/operator/configobservation/images/observe_images.go

```diff
@@ -72,9 +72,11 @@ func ObserveExternalRegistryHostnames(genericListers configobserver.Listers, exi
 	if err != nil {
 		return prevObservedConfig, append(errs, err)
```

Comment:
updated

Reply:
`ObserveInternalRegistryHostname` does the same check.  Anyway, we're handed an error value, and types being a stronger contract than comments, I prefer to err on the side of caution.  (Also, the assignment needs to happen outside the `if` statement so that `o` is in scope when we need it in subsequent statements.)

Reply:
> ObserveInternalRegistryHostname does the same check.

yes i removed it everywhere in my update.

> Anyway, we're handed an error value, and types being a stronger contract than comments, I prefer to err on the side of caution.

I don't want to continue to debate this here, we need this PR merged to re-enable the behavior that was disabled/broken.  Best practices for using the observe pattern, error handling, and unstructured need to be provided by @openshift/sig-master and can be applied as a followup where needed.

Reply:
Sorry, Ben, I didn't see your comment before I posted mine.  I agree, let's go with what you have now rather than debate further.